app/routes/mobile.js

```diff
@@ -0,0 +1,44 @@
+import Route from '@ember/routing/route';
+import ENV from 'website-my/config/environment';
+import { inject as service } from '@ember/service';
+import { toastNotificationTimeoutOptions } from '../constants/toast-notification';
+import { USER_STATES } from '../constants/user-status';
+import redirectAuth from '../utils/redirect-auth';
+
+const API_BASE_URL = ENV.BASE_API_URL;
+
+export default class MobileRoute extends Route {
+  @service toast;
+
+  model = async () => {
+    try {
+      const response = await fetch(`${API_BASE_URL}/users/status/self`, {
+        credentials: 'include',
+      });
+      const userData = await response.json();
+      if (response.status === 200) {
+        return {
+          status: userData?.data?.currentStatus?.state ?? USER_STATES.DNE,
+          userId: userData.userId,
+        };
+      } else if (response.status === 401) {
+        this.toast.error(
+          'You are not logged in. Please login to continue.',
+          '',
+          toastNotificationTimeoutOptions
+        );
+        // added setTimeout here because before new page opens user should be notified of error by toast
+        setTimeout(redirectAuth, 2000);
+      } else if (response.status === 404) {
+        this.toast.error(
+          `Your Status data doesn't exist yet. Please choose your status from the options below.`,
+          '',
+          toastNotificationTimeoutOptions
+        );
+        return { status: USER_STATES.DNE, userId: userData.userId };
+      }
```

Comment:
I don't think we need the `/users/status/self` call here. We can simply call `/users/self` to check the `Unauthorized` state (401) and `Success` (200) state and handle it accordingly (for reference line 24 to 31)
Rest all logic can be removed as of now as we don't need `status` data or userId for `/mobile`
Can be added if there is a need in future

Reply:
You can grab that from /users/self call only

Reply:
done  in last commit .